### PR TITLE
Upgrade Spark to 3.1.2, pom.xml dependencies, add local test script

### PR DESCRIPTION
**Pull request by gyng**
For `basisai/workload-standard:v0.2.3`

- Python 3.8.10
- Spark 3.1.2
- hadoop-aws 3.2.0
- guava 30.1.1
- spark-core_2.12 3.1.2

Adds a simply test for local usage. Need help verifying that this doesn't break anything: I think the local test doesn't cover all the cases.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -36,7 +36,7 @@
 		<dependency>
 			<groupId>com.google.cloud.bigdataoss</groupId>
 			<artifactId>gcs-connector</artifactId>
-			<version>hadoop3-1.9.18</version>
+			<version>1.9.4-hadoop3</version>
 			<!-- Can be removed after upgrading to hadoop 3.3.0 -->
```
Comment (gyng) [inline pom.xml:40]: Don't see a hadoop 3.3.0, did I miss something?

Comment (sweatybridge) [reply pom.xml:40]: we need to keep this version for now. their maven repo is messed up (latest version is not actually latest).

File: Dockerfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
-FROM python:3.8.5-buster
+FROM python:3.9.5-buster
```
Comment (sweatybridge) [inline Dockerfile:1]:
```suggestion
FROM python:3.8.10-buster
```

can we stay on 3.8 for the time being? some ds libraries still don't work with 3.9 (like streamlit)

Comment (gyng) [reply Dockerfile:1]: >🐍 Streamlit can now be installed on Python 3.9. Streamlit components are not yet compatible with Python 3.9 and must use version 3.8 or earlier.

I'll leave it at 3.8 for now though.

File: pom.xml [hunk 3]
```diff
@@ -36,7 +36,7 @@
 		<dependency>
 			<groupId>com.google.cloud.bigdataoss</groupId>
 			<artifactId>gcs-connector</artifactId>
-			<version>hadoop3-1.9.18</version>
+			<version>1.9.4-hadoop3</version>
```
Comment (sweatybridge) [inline pom.xml:39]:
```suggestion
			<version>hadoop3-1.9.18</version>
```

File: pom.xml [hunk 4]
```diff
@@ -58,14 +58,14 @@
 		<dependency>
 			<groupId>org.apache.hadoop</groupId>
 			<artifactId>hadoop-aws</artifactId>
-			<version>3.2.0</version>
+			<version>3.3.0</version>
```
Comment (sweatybridge) [inline pom.xml:61]:
```suggestion
			<version>3.2.0</version>
```

**Comment by sweatybridge** [comment]
Until spark ships a higher hadoop 3.3 in its binary download (currently `spark-3.1.2-bin-hadoop3.2.tgz`), we need to keep the existing gcs and aws connector jar versions. Alternatively we can download spark without hadoop and install our own hadoop 3.3, but it's tedious..

**Review by sweatybridge — APPROVED** [review]

**Comment by gyng** [comment]
Should we disable dependabot if if we don't have a way to check that these dependency upgrades don't break? AFAICT they all passed "CI".

**Comment by sweatybridge** [comment]
Created a chore to look into ci test for spark: https://basis-ai.atlassian.net/browse/BDRK-3297

**Pull request merged**